gems/aws-sdk-docdb/lib/aws-sdk-docdb/plugins/cross_region_copying.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'aws-sigv4'
+
+module Aws
+  module DocDB
+    module Plugins
+      # This plugin populates the `:pre_signed_url` request param for APIs
+      # that support cross region copying.
+      #
+      # This parameter is required by RDS when copying an encrypted snapshot
+      # across regions. This plugin will be skipped if the `:pre_signed_url`
+      # parameter is provided by the user.
+      class CrossRegionCopying < Seahorse::Client::Plugin
+        # @api private
+        class Handler < Seahorse::Client::Handler
+          def call(context)
+            params = context.params
+            if params.is_a?(Hash) &&
```

Comment:
If both `source_region` and `pre_signed_url` are present, this will break right?

Reply:
no, it shouldn't. the customization "models" source region and is treated as valid by the param validator. there are tests for this case for each operation on each service.